app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
in the abstract class, define a method called 'topic' or 'conditional' which fails, then in the actual class define conditional? or call it get_conditional?